geomstats/geometry/spd_matrices.py

```diff
@@ -711,12 +714,109 @@ def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
         inner_product : array-like, shape=[...,]
             Inner-product.
         """
-        spd_space = self.space
-        modified_tangent_vec_a =\
-            spd_space.inverse_differential_power(2, tangent_vec_a, base_point)
-        product = gs.einsum(
-            '...ij,...jk->...ik', modified_tangent_vec_a, tangent_vec_b)
-        result = gs.trace(product, axis1=-2, axis2=-1) / 2
+        eigvals, eigvecs = gs.linalg.eigh(base_point)
+        transp_eigvecs = gs.einsum('...ij->...ji', eigvecs)
```

Comment:
Maybe use `Matrices.transpose(eigvecs)` to improve the readability of the code for someone unfamiliar with einsum